api/src/main/java/org/openmrs/module/sync/api/db/hibernate/HibernateSyncDAO.java

```diff
@@ -1043,14 +1043,12 @@ public void generateDataFile(File outFile, String[] ignoreTables) {
 		}
 		
 	}
-	
-	/**
-	 * Auto generated method comment
-	 * 
-	 * @return
-	 */
+
 	private String[] getConnectionProperties() {
-		Properties props = Context.getRuntimeProperties();
+		return getConnectionProperties(Context.getRuntimeProperties());
+	}
+
+	public String[] getConnectionProperties(Properties props) {

```

Comment:
This change allowed me to get rid of the annoying and impossible to use static PowerMocking in the unit test, and to greatly simplify the testing.